packages/prover/src/proof_provider/proof_provider.ts

```diff
@@ -0,0 +1,205 @@
+
+    const headSlot = this.lightClient.getHead().beacon.slot;
+    await this.lightClient?.sync(
+      computeSyncPeriodAtSlot(headSlot - MAX_REQUEST_LIGHT_CLIENT_UPDATES),
+      computeSyncPeriodAtSlot(headSlot)
+    );
+
+    // Load the payloads from the CL
+    const {start, end} = await getSlotLimitsForPayloads(this.lightClient);
+    const payloads = await getExecutionPayloads(this.options.api, start, end);
+    for (const [slot, payload] of Object.entries(payloads)) {
+      const blockNumber = numberToHex(payload.blockNumber);
+      this.payloads.set(blockNumber, payload);
+      this.slotsMap[parseInt(slot)] = blockNumber;
+    }
+
+    // Load the finalized payload from the CL
+    const finalizedSlot = this.lightClient.getFinalized().beacon.slot;
+    const finalizedPayload = await getExecutionPayloads(this.options.api, finalizedSlot, finalizedSlot);
+    const finalizedBlockNumber = numberToHex(finalizedPayload[finalizedSlot].blockNumber);
+    this.finalizedSlot = finalizedSlot;
+    this.slotsMap[finalizedSlot] = finalizedBlockNumber;
+  }
+
+  getStatus(): {latest: number; finalized: number; status: RunStatusCode} {
+    assertLightClient(this.lightClient);
+
+    return {
+      latest: this.lightClient.getHead().beacon.slot,
+      finalized: this.lightClient.getFinalized().beacon.slot,
+      status: this.lightClient.getStatus(),
+    };
+  }
+
+  getExecutionPayload(blockNumber: number | string | "finalized" | "latest"): allForks.ExecutionPayload {
+    assertLightClient(this.lightClient);
+
+    if (typeof blockNumber === "string" && blockNumber === "finalized") {
+      const payload = this.payloads.get(this.slotsMap[this.finalizedSlot]);
+      if (!payload) throw new Error("No finalized payload");
+      return payload;
+    }
+
+    if (typeof blockNumber === "string" && blockNumber === "latest") {
+      const payload = this.payloads.last;
+      if (!payload) throw new Error("No latest payload");
+      return payload;
+    }
+
+    if (typeof blockNumber === "string" && blockNumber.startsWith("0x")) {
+      const payload = this.payloads.get(blockNumber);
+      if (!payload) throw new Error(`No payload for blockNumber ${blockNumber}`);
+      return payload;
+    }
+
+    if (typeof blockNumber === "number") {
+      const payload = this.payloads.get(numberToHex(blockNumber));
+      if (!payload) throw new Error(`No payload for blockNumber ${blockNumber}`);
+      return payload;
+    }
+
+    throw new Error(`Invalid blockNumber "${blockNumber}"`);
+  }
+
+  async update(lcHeader: allForks.LightClientHeader, finalized = false): Promise<void> {
+    const fork = this.options.config.getForkName(lcHeader.beacon.slot);
+
+    if (!isForkExecution(fork)) {
+      return;
+    }
+
+    if (isForkExecution(fork) && !("execution" in lcHeader)) {
+      throw new Error("Execution payload is required for execution fork");
+    }
+
+    const newPayload = (lcHeader as capella.LightClientHeader).execution;
```

Comment:
rename it to newPayloadHeader for clarity as this is not a full payload with transactions